cli/src/semdep/package_restrictions.py

```diff
@@ -35,14 +34,13 @@ def semver_matches(expression: str, actual_version: str) -> bool:
 # compare vulnerable range to version in lockfile
 def dependencies_range_match_any(
     search_for_ranges: List[ProjectDependsOnEntry],
-    lockfile_path: Path,
     have_deps: List[LockfileDependency],
-) -> Generator[Tuple[ProjectDependsOnEntry, LockfileDependency, Path], None, None]:
+) -> Generator[Tuple[ProjectDependsOnEntry, LockfileDependency], None, None]:
```

Comment:
```suggestion
) -> Iterator[Tuple[ProjectDependsOnEntry, LockfileDependency]]:
```

I forget how generators are ever useful but I think you typically use them when you want to communicate some more input to it while it's iterating